### PR TITLE
[Feature] Refactor create garden page

### PR DESCRIPTION
**Pull request by daiHash**
## Description
- Refactor create garden page
- Use api helper
- Prevent multiple calls
- Add loading
- Remove rule

## Closes issue(s)
#123 

## Screenshots
refactor-create

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [x] I have tested this code

### CODE REVIEW

File: src/pages/CreateGarden/components/AddRules.tsx [hunk 1]
```diff
@@ -119,6 +125,15 @@ export const AddRules: React.FC<AddRulesProps> = ({
             {userRules.map((rule, idx) => (
               <ListItem key={`${rule.name}-${idx}`} className={classes.ruleLi}>
                 <UserRule name={rule.name} description={rule.description} />
+                {idx}
+                <div
+                  className={classes.closeIcon}
+                  onClick={() => {
+                    removeRule(idx);
+                  }}
+                >
```
Comment (daiHash) [inline src/pages/CreateGarden/components/AddRules.tsx:134]: Remove rule

File: src/pages/CreateGarden/components/GardenSummary.tsx [hunk 2]
```diff
@@ -134,15 +148,21 @@ export const GardenSummary: React.FC<GardenSummaryProps> = ({
...
+          color="primary"
+          onClick={createGardenHandler}
+          startIcon={<AddCircleIcon />}
+        >
+          Add Flower Bed
+        </Button>
+        {loading && (
+          <CircularProgress size={28} className={classes.buttonProgress} />
+        )}
+      </div>
```
Comment (daiHash) [inline src/pages/CreateGarden/components/GardenSummary.tsx:165]: Disable button when calling api  and show loading

File: src/pages/CreateGarden/index.tsx [hunk 3]
```diff
@@ -95,43 +115,51 @@ export const CreateGarden = () => {
...
+  useEffect(() => {
+    if (createGardenApi.status === "succeeded") {
+      const { _id: gardenId } = createGardenApi.response;
+      if (gardenId) {
+        createGardenRulesBulk(gardenId);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [createGardenApi.response, createGardenApi.status]);
+
```
Comment (daiHash) [inline src/pages/CreateGarden/index.tsx:162]: Refactor using api helper

File: src/pages/CreateGarden/index.tsx [hunk 4]
```diff
@@ -209,7 +239,7 @@ export const CreateGarden = () => {
                 color="primary"
                 startIcon={<ArrowLeftIcon />}
                 onClick={handleBack}
-                disabled={activeStep === 0}
+                disabled={activeStep === 0 || isApiProcessing}
               >
```
Comment (daiHash) [inline src/pages/CreateGarden/index.tsx:243]: Prevent going back in the form when waiting for api response

File: mockServer/server.ts [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
 
 app.use(cors());
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
-app.use("/api/v1/tmp", tmpRouter);
 app.use("/api/v1/gardens", gardensRouter);
+app.use("/api/v1/rules", rulesRouter);
 
 app.listen(3001, () => {
```
Comment (daiHash) [inline mockServer/server.ts:15]: Added currently using api mock data, so please try to use mock server for dev and add mock data when adding a new api. (need to change the env var for api base url to http://localhost:3001/api/v1)

File: src/models/rule.model.tsx [hunk 6]
```diff
@@ -3,6 +3,7 @@ export interface Rule {
   name: string;
   description: string;
   gardenId: string;
+  createdDate: string;
+  lastUpdate: string;
+  IsRemoved: boolean;
```
Comment (kome12) [inline src/models/rule.model.tsx:8]: Probably should be `isRemoved`

**Review by kome12 — COMMENTED** [review]
@daiHash Overall, looks good to me! Just 1 thing, I think the rule.model has a capital field but probably it should be camel case.

**Review by kome12 — APPROVED** [review]
LGTM!

**Pull request merged**